lib/promiscuous/backend/poseidon.rb

```diff
@@ -0,0 +1,128 @@
+require 'poseidon'
+require 'poseidon_cluster'
+
+class Promiscuous::Backend::Poseidon
+  attr_accessor :connection, :connection_lock
+
+  def initialize
+    # The poseidon socket doesn't like when multiple threads access to it apparently
+    @connection_lock = Mutex.new
+  end
+
+  def new_connection
+    client_id = ['promiscuous', Promiscuous::Config.app, Poseidon::Cluster.guid].join('.')
+    @connection = ::Poseidon::Producer.new(Promiscuous::Config.kafka_hosts, client_id,
+                                          :type => :sync,
+                                          :compression_codec => :none,
+                                          :metadata_refresh_interval_ms => 600_000,
+                                          :max_send_retries => 10,
+                                          :retry_backoff_ms => 100,
+                                          :required_acks => 1,
+                                          :ack_timeout_ms => 1000,
+                                          :socket_timeout_ms => Promiscuous::Config.socket_timeout)
+    @connection
+  end
+
+  def connect
+    @connection = new_connection
+  end
+
+  def disconnect
+    @connection_lock.synchronize do
+      return unless connected?
+      @connection.shutdown
+      @connection = nil
+    end
+  end
+
+  # TODO: extend Poseidon with a connected? method
+  def connected?
+    !@connection.nil?
+  end
+
+  def raw_publish(options)
+    ok = @connection.send_messages([Poseidon::MessageToSend.new(options[:topic], options[:payload], options[:key])])
+    raise "Unable to send messages" if !ok
+    Promiscuous.debug "[publish] [kafka] #{options[:topic]}/#{options[:topic_key]} #{options[:payload]}"
+  end
+
+  def publish(options={})
+    @connection_lock.synchronize do
+      raw_publish(options)
+      # TODO: Rabbit is in control of the unlocking for now
```

Comment:
This was a bad comment and will be addressed with the `:both` backend i am working